likecoin/js/sidebar/src/pages/LikeCoinPluginSideBar.js

```diff
@@ -29,12 +29,24 @@ function LikeCoinPluginSideBar(props) {
   const [showFilledMattersTickBox, setShowFilledMattersTickBox] = useState(false);
   const [showMattersDraftLink, setShowMattersDraftLink] = useState(false);
   const [showMattersArticleLink, setShowMattersArticleLink] = useState(false);
+  const [ISCNVersionString, setISCNVersionString] = useState(true);
+  const [showUpdateISCNButton, setShowUpdateISCNButton] = useState(true);
   const [pinBarIconColor, setPinBarIconColor] = useState('#28646E');
   const [distributeToMatters, setDistributeToMatters] = useState(true);
   const isPluginSidebarOpened = useSelect((select) => select('core/edit-post')
     .isPluginSidebarOpened());
   const isCurrentPostPublished = useSelect((select) => select('core/editor')
     .isCurrentPostPublished());
+  const postDate = useSelect((select) => select('core/editor').getEditedPostAttribute('modified_gmt'));
+  useEffect(() => {
+    setShowUpdateISCNButton(isCurrentPostPublished
+      && props.ISCNTimestamp
+      && Date.parse(`${postDate}Z`) > props.ISCNTimestamp); // force parsing as gmt
+  }, [isCurrentPostPublished, postDate, props.ISCNTimestamp]);
```

Comment:
I see this part of code appear in [LikeCoinPluginDocumentSettingPanel.js](https://github.com/likecoin/likecoin-wordpress/pull/176/files?diff=unified&w=1#diff-deb69af46aa4990a95807283ab2fea62f077a514bc4b9eb119bc8eb85149d50a), I think this can be refactored into a custom hook in the future